src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,43 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, V> {
+    private T first;
+    private V second;
+
+    public Pair(T t, V v) {
+        this.first = t;
+        this.second = v;
+    }
+
+    public T getFirst() {
+        return this.first;
+    }
+
+    public V getSecond() {
+        return this.second;
+    }
+
+    public static <T, V> Pair<T, V> of(T first, V second) {
+        return new Pair(first, second);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Pair)) {
+            return false;
+        }
+        Pair<T, V> pair = (Pair) o;
+        return pair.getFirst().equals(this.first)
+                && pair.getSecond().equals(this.second);
+    }
+
+    @Override
+    public int hashCode() {
+        return this.getFirst().hashCode() * 23 + this.getSecond().hashCode() / 31;
```

Comment:
```suggestion
        return this.getFirst().hashCode() * 23 + this.getSecond().hashCode() * 31;
```

I don't think that division by 31 is a good idea? because if this.getSecond().hashCode() == 30 or this.getSecond().hashCode() == 21, then the division by 31 in both cases will be 0. This is bad practice. You should understand how it works, not just type any symbols